File: gestion-projets-frontend/src/app/models/utilisateur.ts
export enum Role {
  ADMIN = 'ADMIN',
  UTILISATEUR = 'UTILISATEUR'
}

export interface Utilisateur {
  id?: number;
  email: string;
  nom: string;
  prenom: string;
  motDePasse?: string;
  role: Role;
  actif: boolean;
  dateCreation?: Date;
}

export interface LoginRequest {
  email: string;
  motDePasse: string;
}

export interface LoginResponse {
  success: boolean;
  message: string;
  utilisateur: Utilisateur;
}

File: GestionProjets/backend/gestion-projets/src/main/java/com/example/gestion_projets/entity/Utilisateur.java
package com.example.gestion_projets.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Utilisateur {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String email;

    @Column(nullable = false)
    private String nom;

    @Column(nullable = false)
    private String prenom;

    @Column(nullable = false)
    private String motDePasse;

    @Enumerated(EnumType.STRING)
    private Role role = Role.UTILISATEUR;

    private boolean actif = true;

    private LocalDateTime dateCreation;

    @PrePersist
    protected void onCreate() {
        dateCreation = LocalDateTime.now();
    }
}